src/dbnode/storage/index.go

```diff
@@ -1064,8 +1017,6 @@ func (i *nsIndex) Query(
 	}
 
 	results.Lock()
-	// Signal not to add any further results since we've returned already.
-	results.returned = true
```

Comment:
Are we losing the ability to do this signaling? Seems like this might make the impact of timed out queries even worse since they'll keep updating this map...Actually isn't it broken because they'll keep trying to add results to a map that may have been returned to the pool?

Reply:
So I now use a lifetime to protect against writing to the results once we return from the Query call to the index, this prevents writing to results during cancellation or any other early return code path.